src/components/menu/menu.component.js

```diff
@@ -1,8 +1,74 @@
-import React from 'react';
+import React, { useRef } from 'react';
 import PropTypes from 'prop-types';
 import { StyledMenuWrapper, StyledMenuItemsWrapper, StyledMenuItem } from './menu.style';
+import Events from '../../utils/helpers/events';
 
 const Menu = ({ menuType = 'light', children }) => {
+  const menuItemsRefs = useRef(React.Children.map(children, child => child.ref || React.createRef()));
+  let actualIndex;
+  const setFocusToElement = (event, index) => {
+    if (event) {
+      event.preventDefault();
+    }
+
+    let newIndex = index;
+    actualIndex = index;
+    if (index === menuItemsRefs.current.length) {
```

Comment:
As before, please add more descriptive variables for the `if` condition and the `index` variables.

Reply:
I removed `newIndex` and `actualIndex` and replaced it all with `actualFocusedItemIndex`